src/formatMarkup.js

```diff
@@ -120,13 +129,8 @@ export const diffableFormatter = function (markup, options) {
     // <tags and="attributes" />
     let result = '\n' + '  '.repeat(indent) + '<' + node.nodeName;
 
-    let endingAngleBracket = '>';
-    if (
-      tagIsVoidElement &&
-      options.voidElements === 'xhtml'
-    ) {
-      endingAngleBracket = ' />';
-    }
+    const shouldSelfClose = (tagIsVoidElement && options.voidElements === 'xhtml') || (options.selfClosingTag && !hasChildren && !tagIsEscapabelRawTextElement);
+    const endingAngleBracket = shouldSelfClose ? ' />' : '>';
```

Comment:
This is much harder to read now. Always break up your code so there is **ONE IDEA PER LINE**.

If you want to extract the condition and give it a variable name, that's fine though.

```suggestion
    const shouldSelfClose = (
      (
        tagIsVoidElement &&
        options.voidElements === 'xhtml'
      ) ||
      (
        options.selfClosingTag &&
        !hasChildren &&
        !tagIsEscapabelRawTextElement
      )
    );
    let endingAngleBracket = '>';
    if (shouldSelfClose) {
     endingAngleBracket = ' />';
    }
```